internal/cli/terraform_fetcher.go

```diff
@@ -0,0 +1,79 @@
+package cli
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/auth0/go-auth0/management"
+
+	"github.com/auth0/auth0-cli/internal/auth0"
+)
+
+type (
+	importDataList []importDataItem
+
+	importDataItem struct {
+		ResourceName string
+		ImportID     string
+	}
+
+	resourceDataFetcher interface {
+		FetchData(ctx context.Context) (importDataList, error)
+	}
+
+	clientResourceFetcher struct {
+		api *auth0.API
+	}
+)
+
+func (f *clientResourceFetcher) FetchData(ctx context.Context) (importDataList, error) {
+	var data importDataList
+
+	var page int
+	for {
+		clients, err := f.api.Client.List(
+			ctx,
+			management.Page(page),
+			management.Parameter("is_global", "false"),
+			management.IncludeFields("client_id", "name"),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, client := range clients.Clients {
+			data = append(data, importDataItem{
+				ResourceName: "auth0_client." + sanitizeResourceName(client.GetName()),
+				ImportID:     client.GetClientID(),
+			})
+		}
+
+		if !clients.HasNext() {
+			break
+		}
+
+		page++
+	}
+
+	return data, nil
+}
+
+// sanitizeResourceName will return a valid terraform resource name.
+//
+// A name must start with a letter or underscore and may
+// contain only letters, digits, underscores, and dashes.
+func sanitizeResourceName(name string) string {
+	// Regular expression pattern to remove invalid characters.
+	namePattern := "[^a-zA-Z0-9_-]+"
+	re := regexp.MustCompile(namePattern)
+
+	sanitizedName := re.ReplaceAllString(name, "")
+
+	// Regular expression pattern to remove leading digits or dashes.
+	namePattern = "^[0-9-]+"
```

Comment:
Would it be preferable to match any leading character that is not a letter or underscore?

Reply:
But it's only matching a subset of invalid characters, not all possible invalid characters.

Reply:
Ah, you're already removing all the other possible chars in the step above.

Reply:
The logic is split in 2 through 2 regexes, the first one removes all invalid characters, and the second one removes digits or - from the start of the string.

Reply:
Thanks for the explanation. My bad.

Reply:
No worries, happy to clarify 👍🏻